beluga_ros/include/beluga_ros/ndt_ellipsoid.hpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_ROS_NDT_ELLIPSOID_HPP
+#define BELUGA_ROS_NDT_ELLIPSOID_HPP
+
+#include <beluga_ros/messages.hpp>
+
+/**
+ * \file
+ * \brief Utilities for NDT I/O over ROS interfaces.
+ */
+
+namespace beluga_ros {
+
+static void use_mean_covariance(const auto& mean, const auto& eigenSolver, auto& marker){
+  marker.type = beluga_ros::msg::Marker::SPHERE;
+  marker.action = beluga_ros::msg::Marker::ADD;
+  
+  // Compute the rotation based on the covariance matrix
+  const auto eigenvectors = eigenSolver.eigenvectors().real();
+  const auto eigenvalues = eigenSolver.eigenvalues();
+
+  Eigen::Matrix3d rotationMatrix;
+  Eigen::Vector3d scalevector;
+  rotationMatrix << eigenvectors.col(0), eigenvectors.col(1), eigenvectors.col(2);
+  scalevector = Eigen::Vector3d{eigenvalues.x(), eigenvalues.y(), eigenvalues.z()};
+
+  // Permutation
+  if (std::abs(rotationMatrix.determinant() - 1.0) > Eigen::NumTraits<double>::dummy_precision()) {
+    rotationMatrix << eigenvectors.col(1), eigenvectors.col(0), eigenvectors.col(2);
+    scalevector = Eigen::Vector3d{eigenvalues.y(), eigenvalues.x(), eigenvalues.z()};
+  }
+
+  Eigen::Quaterniond rotation{1.0f, 0.0f, 0.0f, 0.0f};
+  if (std::abs(rotationMatrix.determinant() - 1.0) < Eigen::NumTraits<double>::dummy_precision()) {
+    rotation = Eigen::Quaterniond{rotationMatrix};
+  }
```

Comment:
@Fernando-Sanz meta: are there any instances of non-orthogonal eigenvectors? They should be for a symmetric covariance matrix. If it's a numerical issue, I'd suggest we orthogonalize rather than ignore.

Reply:
I think we can probably just ignore this case (?).
Or just warn if it happens, it shouldn't happen in practice.

Or do the same as when the eigenvector solver does not converge and draw a cell-size square